pytools/test/test_persistent_dict.py

```diff
@@ -432,9 +432,13 @@ def test_scalar_hashing() -> None:
     assert keyb(np.int32(1)) == keyb(np.int32(1))
     assert keyb(np.int32(2)) != keyb(np.int32(1))
     assert keyb(np.int64(1)) == keyb(np.int64(1))
-    assert keyb(1) == keyb(np.int64(1))
+    assert keyb(1) != keyb(np.int64(1))
     assert keyb(1) != keyb(np.int32(1))
 
+    assert keyb(1) == keyb("1")
+    assert keyb(1.0) != keyb("1.0")  # '1.0' uses hex representation
+    assert keyb(1+1j) == keyb("(1+1j)")
```

Comment:
Not sure if this is an issue. If it is, we could change the KeyBuilder to `(int, str(key)`